### PR TITLE
Submitting Lesson 02 exercise for Mark L Taylor 

### PR DESCRIPTION
**Pull request by mark-l-taylor**
grid_printer.py - prints grids in a variety of sizes
	consists of three parts, each part is the progress of the code with part03 printing all three types of grids.
fizzbuzz.py - iterates on numbers and prints based on divisible by 3 or 5.
series.py - returns the value of the nth number in a series (fibonacci,lucas, etc.)

### CODE REVIEW

**Comment by mark-l-taylor** [comment]
My pull request from a couple days ago has not been approved yet.  If I understand what is happening I am amending the pull request to include the additional commits for lesson 03 exercises.

I have merged with the latest remote master and hopefully submitting this correctly

File: students/mark-l-taylor/lesson02/fizzbuzz.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    line = ''
+    if i % 3 == 0:
+        line += 'Fizz'
+    if i % 5 == 0:
+        line += 'Buzz'
+    #Based on result of test either print line or the number
+    if line:
+        print(line)
+    else:
+        print(i)
```
Comment (charlesrobison) [inline students/mark-l-taylor/lesson02/fizzbuzz.py:19]: This works

File: students/mark-l-taylor/lesson02/grid_printer_part01.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+#+ - - - - + - - - - +
+ 
+
+#Define two different lines
+full_line = '+ ' + 4*'- ' + '+ ' + 4*'- ' + '+'
+side_line = '|' + 9*' ' + '|' + 9*' ' + '|'
+
+#Print the lines to create the grid
+for i in range(0,3):
+	print(full_line)
```
Comment (charlesrobison) [inline students/mark-l-taylor/lesson02/grid_printer_part01.py:21]: Over indented here - make sure you set your editor to indent 4 spaces

File: students/mark-l-taylor/lesson02/grid_printer_part02.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        for j in range(0,n//2):
+            if i < 2:
+                side_line(n)
+            else:
+                pass
+                
+#Actually print the grids
+print_grid(9)
+print_grid(3)
+print_grid(15)
```
Comment (charlesrobison) [inline students/mark-l-taylor/lesson02/grid_printer_part02.py:36]: Good

File: students/mark-l-taylor/lesson02/grid_printer_part03.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+'''Grid Printer
+    Print grids based on user inputs.
+'''
+ 
+
+def full_line(m=2,n=9):
```
Comment (charlesrobison) [inline students/mark-l-taylor/lesson02/grid_printer_part03.py:6]: Need a space after the comma

Comment (charlesrobison) [reply students/mark-l-taylor/lesson02/grid_printer_part03.py:6]: Here, as well as in the functions below

File: students/mark-l-taylor/lesson02/grid_printer_part03.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+print('Basic Grid (Part 1)')
+print_grid()
+
+print('\nThree different grids (Part 2)')
+print_grid(9)
+print_grid(3)
+print_grid(15)
+
+print('\nGrids of different sizes (Part 3)')
+print_grid2(3,4)
```
Comment (charlesrobison) [inline students/mark-l-taylor/lesson02/grid_printer_part03.py:50]: Spaces after commas needed here too

File: students/mark-l-taylor/lesson02/grid_printer_part03.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+print_grid2(5,3)
```
Comment (charlesrobison) [inline students/mark-l-taylor/lesson02/grid_printer_part03.py:51]: Works well!

File: students/mark-l-taylor/lesson02/series.py [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+    # test if sum_series works for arbitrary initial values
+    assert sum_series(0, 3, 2) == 3
+    assert sum_series(1, 3, 2) == 2
+    assert sum_series(2, 3, 2) == 5
+    assert sum_series(3, 3, 2) == 7
+    assert sum_series(4, 3, 2) == 12
+    assert sum_series(5, 3, 2) == 19
+
+    print("tests passed")
```
Comment (charlesrobison) [inline students/mark-l-taylor/lesson02/series.py:76]: Nice!

File: students/mark-l-taylor/lesson03/list_lab.py [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+fruit = ['Apples', 'Pears', 'Oranges', 'Peaches']
+print(fruit)
+
+#Ask the user for another fruit
+response = input('Please add another fruit to the list:')
+fruit.append(response)
+print(fruit)
+
+#Ask user for a number and return which fruit
+response = input('Select a number between 1 and {}:'.format(len(fruit)))
```
Comment (charlesrobison) [inline students/mark-l-taylor/lesson03/list_lab.py:17]: Good!

File: students/mark-l-taylor/lesson03/list_lab.py [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+print(fruit)
+
+#Add another fruit to the beginning of list using insert
+fruit.insert(0,'Strawberries')
+print(fruit)
+
+#Display all the fruit that begins with "P" using for loop
+print('\nFruit in list that start with "P"')
+for f in fruit:
+    if f.startswith('P'):
```
Comment (charlesrobison) [inline students/mark-l-taylor/lesson03/list_lab.py:31]: You can use upper() or similar to catch case sensitive user input

**Review by charlesrobison — APPROVED** [review]
Nice job on these - just a few comments.  Let me know if you have questions.

Ready to merge

**Pull request merged**